topcoffea/modules/YieldTools.py

```diff
@@ -0,0 +1,424 @@
+    def get_scaled_yield(self,hin_dict,year,proc,cat,njets_cat,overflow_str,rwgt_pt=None):
+
+        h = hin_dict["njets"]
+
+        # Figure out which njet bins we want to combine
+        if isinstance(njets_cat,str):
+            njet = self.JET_BINS[njets_cat][0]
+        elif isinstance(njets_cat,int):
+            njet = njets_cat
+        else:
+            raise Exception(f"Wrong type, expected str or int, but got a {type(njets_cat)}, exiting...")
+
+        # Integrate over the categories for the analysis category we are interested in
+        h = self.select_hist_for_ana_cat(h,self.CATEGORIES[cat],njet)
+
+        # Reweight the hist
+        if rwgt_pt is not None:
+            hist.set_wilson_coefficients(**wc_vals)
+        else:
+            h.set_sm()
+
+        # Scale by sum of weights (if EFT hist)
+        h_sow = hin_dict["SumOfEFTweights"]
+        nwc = h_sow._nwc
+        if nwc > 0:
+            h.set_sm()
+            sow_val , sow_err = self.get_yield(h_sow,proc)
+            h.scale(1.0/sow_val) # Divide EFT samples by sum of weights at SM, ignore error propagation for now
+
+        # Scale by lumi
+        lumi = 1000.0*self.get_lumi(year)
+        h.scale(lumi)
+
+        return self.get_yield(h,proc,overflow_str)
+
+
+    # This function:
+    #   - Takes as input a hist dict (i.e. what the processor outptus)
+    #   - Retruns a dictionary of yields for the categories in CATEGORIES
+    #   - Making use of get_scaled_yield()
+    #   - If you pass a key from JET_BINS for yields_for_njets_cats, will make a dict of yields for all the jet bins in that lep cat
+    def get_yld_dict(self,hin_dict,year,yields_for_njets_cats=None):
+
+        yld_dict = {}
+        proc_lst = self.get_cat_lables(hin_dict,"sample")
+        for proc in proc_lst:
+            proc_name_short = self.get_short_name(proc)
+            yld_dict[proc_name_short] = {}
+            for cat,cuts_dict in self.CATEGORIES.items():
+                if "2lss" in cat: njet_cat = "2lss"
+                elif "3l" in cat: njet_cat = "3l"
+                elif "4l" in cat: njet_cat = "4l"
+
+                # We want to sum over the jet bins, and look at all of the lep cats
+                if yields_for_njets_cats is None:
+                    yld_dict[proc_name_short][cat] = self.get_scaled_yield(hin_dict,year,proc,cat,njet_cat,overflow_str="over")
+
+                # We want to look at all the jet bins in the give lep cat
+                elif yields_for_njets_cats == njet_cat:
+                    for njet in self.JET_BINS[njet_cat]:
+                        if njet == max(self.JET_BINS[njet_cat]): include_overflow = "over"
+                        else: include_overflow = "none"
+                        cat_name_full = cat+"_"+str(njet)+"j"
+                        yld_dict[proc_name_short][cat_name_full] = self.get_scaled_yield(hin_dict,year,proc,cat,njet,overflow_str=include_overflow)
+
```

Comment:
No `else` check here to catch potentially invalid inputs for `yields_for_njets_cats`?

Reply:
Thanks, I've added [a check](https://github.com/TopEFT/topcoffea/commit/4a796681f6413533779a6d938923929051c32612) for that.